List.h

```diff
@@ -0,0 +1,292 @@
+#pragma once
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+
+enum Node { NODE_DATA, NODE_PREV, NODE_NEXT, NODE_SIZE };
+enum Data { DATA_VALUE, DATA_NULL_TERMINATOR, DATA_SIZE };
+
+const int FIRST_GRATER_THAN_SECOND = 0;
+
+
+// Initializes list
+void StringListInit(char*** head_node, const char* str) {
+	if (!*head_node) {
+		*head_node = (char**)calloc(NODE_SIZE, sizeof(char*));
+
+		size_t size_of_str = strlen(str);
+		size_t size_of_str_memory = (size_of_str + DATA_NULL_TERMINATOR) * sizeof(char);
+
+		(*head_node)[NODE_DATA] = (char*)calloc(size_of_str_memory, sizeof(char));

```

Comment:
we do not nee to allocate a new memory if 'str' is null or empty, please verify input parameter 'str' before